webpack.config.js

```diff
@@ -0,0 +1,38 @@
+const path = require('path');
+const nodePath = process.env.NODE_PATH || './node_modules';
+
+module.exports = {
+//    mode: "development",
+    mode: "production",
+    entry: path.resolve(__dirname, './kuma/javascript/src/index.jsx'),
+    output: {
+        filename: 'react.js',
+        path: path.resolve(__dirname, './kuma/javascript/dist/')
+    },
+    module: {
+        rules: [{
+            "test": /\.jsx?$/,
+            "exclude": /node_modules/,
+            "use": {
+                "loader": "babel-loader",
+                "options": {
+                    "presets": [
+                        "@babel/preset-env",
```

Comment:
This and other ``presets`` (and possibly ``plugins``) need a prefix to work in the docker container:

```
nodePath + "/@babel/preset-env"
```
``nodePath`` resolves to ``./node_modules`` when run normally, and ``/tools/node_modules`` in the Docker container.

The ``tools`` directory was used to avoid an issue where node always looks in a local ``.node_modules`` folder first, and this may contain packages built on a macOS or Windows system, which are not compatible with the Debian system in the Docker container.

I wish we had something like Python's virtual environments, so we could instead tell node "don't looks at ``./node_modules``, look here", but as the [old npm FAQ](
https://web.archive.org/web/20150216184318/https://docs.npmjs.com/misc/faq#node_modules-is-the-name-of-my-deitys-arch-rival-and-a-forbidden-word-in-my-religion-can-i-configure-npm-to-use-a-different-folder), this is baked into node and the maintainers aren't planning to ever change it. ``NODE_PATH`` might be part of the solution, but I think even then ``./node_modules`` wins.

You can verify it works locally with:
```
VERSION=latest make build-base build-kuma
```

If this works, then TravisCI and production should work too.

Reply:
John: I took the nodePath stuff out because it was causing my local webpack runs to fail, and I couldn't see why it would be necessary since you also have it set in the resolve and resolveLoader properties of the config.

I can the local webpack again; maybe I was just doing something wrong. But I've never seen a config file that explicitly includes the node_modules/ directory when specifying a dependency, so I'm hoping we can find a different way to make the config work in both contexts.